src/discussionBox/discussionBox.js

```diff
@@ -0,0 +1,259 @@
+import './discussionBox.css';
+import {IconButton, Modal, TextInput, ButtonC, ButtonA} from '../UI';
+import { useNavigate  } from "react-router-dom";
+import { useContext, useState} from 'react';
+import Context from '../../../context';
+import { get, post, put } from '../../../util/api';
```

Comment:
Right now, these components won't work because of their dependencies on outside functions. We can work on making them more generic, but we'll work with what you've provided.